node/node.go

```diff
@@ -546,21 +547,23 @@ func adjustNftTokenIdentifier(token string, nonce uint64) string {
 	return formattedTokenIdentifier
 }
 
-func (n *Node) getAccountHandlerAPIAccounts(address string) (state.UserAccountHandler, error) {
-	componentsNotInitialized := check.IfNil(n.coreComponents.AddressPubKeyConverter()) ||
-		check.IfNil(n.stateComponents.AccountsAdapterAPI())
+func (n *Node) decodeAddressToPubKey(address string) ([]byte, error) {
+	// Question for review: do we still require the special lazy initialization logic?
+	// > other functions seem to not care about this (and directly use the converter). E.g. GetESDTsRoles()
+	componentsNotInitialized := check.IfNil(n.coreComponents.AddressPubKeyConverter())
 	if componentsNotInitialized {
-		return nil, errors.New("initialize AccountsAdapterAPI, PubkeyConverter first")
+		return nil, errors.New("PubkeyConverter not initialized")
 	}
 
-	addr, err := n.coreComponents.AddressPubKeyConverter().Decode(address)
+	pubKey, err := n.coreComponents.AddressPubKeyConverter().Decode(address)
 	if err != nil {
-		return nil, errors.New("invalid address, could not decode from: " + err.Error())
+		return nil, fmt.Errorf("invalid address (%w): %s", err, address)
 	}
 
-	return n.getAccountHandlerForPubKey(addr)
+	return pubKey, nil
 }
 
+// Question for review: this is only used for GetAllIssuedESDTs. Change that ones, as well, then remove this and the dependency on "AccountsAdapterAPI"?
```

Comment:
Yes, we can remove the AccountsAdapterAPI. You could get the latest root hash from node.go and call the GetExistingAccounts with it. Alternatively, the rootHash param can be a `[]byte("latest")` and the accounts repository will know to fetch it

Reply:
Removed the function `getAccountHandlerForPubKey` for now. 

`AccountsAdapterAPI` is still used in `node.go` (sorry for being mistaken earlier) etc. In order to avoid further non-trivial changes, I've added a task, if still applicable: EN-12400.